server/controllers/utils/overviewController/buildOverviewActions.test.ts

```diff
@@ -89,193 +53,197 @@ describe('buildOverviewActions', () => {
-        socNominal,
-        user,
-        staffRoles,
-        config,
-        undefined,
-      )
-      expect(
-        !!resp.find(
-          action =>
-            action.url === `${config.serviceUrls.pathfinder}/refer/offender/${PrisonerMockDataA.prisonerNumber}`,
-        ),
-      ).toEqual(visible)
-    })
+      perms                              | pathfinderNominalToUse | visible
+      ${{ pathfinder: { edit: true } }}  | ${undefined}           | ${true}
+      ${{ pathfinder: { edit: true } }}  | ${pathfinderNominal}   | ${false}
+      ${{ pathfinder: { edit: false } }} | ${undefined}           | ${false}
+    `(
+      'user with pathfinder.edit: $perms.pathfinder.edit with pathfinderNominal: $pathfinderNominalToUse can see refer to pathfinder: $visible',
+      ({ perms, pathfinderNominalToUse, visible }) => {
+        const resp = buildOverviewActions(
+          PrisonerMockDataA,
+          pathfinderNominalToUse,
+          socNominal,
+          prisonUserMock,
+          config,
+          undefined,
+          perms,
+        )
+        expect(
+          !!resp.find(
+            action =>
+              action.url === `${config.serviceUrls.pathfinder}/refer/offender/${PrisonerMockDataA.prisonerNumber}`,
+          ),
+        ).toEqual(visible)
+      },
+    )
   })
 
-  test.each([
-    [[Role.SocCustody], undefined, true],
-    [[Role.SocCommunity], undefined, true],
-    [[Role.SocDataAnalyst], undefined, true],
-    [[Role.SocDataManager], undefined, true],
-    [[], undefined, false],
-    [[Role.SocHq], undefined, false],
-    [[Role.SocCustody], socNominal, false],
-  ])(
-    `user with roles: %s and soc nominal: %s, can see: $s`,
-    (roles: Role[], socNominalToUse: Nominal | undefined, visible: boolean) => {
-      const user = { ...prisonUserMock, userRoles: roles }
-      const resp = buildOverviewActions(
-        PrisonerMockDataA,
-        pathfinderNominal,
-        socNominalToUse,
-        user,
-        staffRoles,
-        config,
-        undefined,
-      )
-      expect(
-        !!resp.find(
-          action =>
-            action.url === `${config.serviceUrls.manageSocCases}/refer/offender/${PrisonerMockDataA.prisonerNumber}`,
-        ),
-      ).toEqual(visible)
-    },
-  )
+  describe('Add to SOC', () => {
+    test.each`
+      perms                       | socNominalToUse | visible
+      ${{ soc: { edit: true } }}  | ${undefined}    | ${true}
+      ${{ soc: { edit: true } }}  | ${socNominal}   | ${false}
+      ${{ soc: { edit: false } }} | ${undefined}    | ${false}
+    `(
+      'user with pathfinder.edit: $perms.soc.edit with socNominal: socNominalToUse can see add to soc: $visible',
+      ({ perms, socNominalToUse, visible }) => {
+        const resp = buildOverviewActions(
+          PrisonerMockDataA,
+          pathfinderNominal,
+          socNominalToUse,
+          prisonUserMock,
+          config,
+          undefined,
+          perms,
+        )
+        expect(
```

Comment:
Since there are so many of these, it might be nice to have a little helper function:

`expectActionVisible(url, visible)`
?

Reply:
Jon now you want me to fix everyone else's work!